src/registryVariants.ts

```diff
@@ -0,0 +1,153 @@
+    strictDirectoryName?: boolean;
+    /**
+     * Whether or not to ignore the fullName that's parsed from the file path. If true, the metadata type's
+     * name will be used instead. For example, CustomLabels instead of MyLabels.
+     */
+    ignoreParsedFullName?: boolean;
+    /**
+     * If the type is a folder type (container for components), the id of the type it is a container for.
+     */
+    folderContentType?: string;
+    /**
+     * If the type is contained in folders, the id of the type that contains it.
+     */
+    folderType?: string;
+    /**
+     * If the parent name should be ignored when constructing the type's fullName
+     */
+    ignoreParentName?: boolean;
+    /**
+     * The XML element name for the type in the xml file used for constructing child components.
+     */
+    xmlElementName?: string;
+    /**
+     * When converting deploying source, this will update the suffix in the output or temporary directory (metadata format)
+     * Use this, along with additional suffix keys in the registry, to support incorrect suffixes from existing code
+     */
+    legacySuffix?: string;
+    /**
+     * The xml attribute used as the unique identifier when parsing the xml
+     */
+    uniqueIdElement?: string;
+    /**
+     * Whether the component is supported by the Metadata API and therefore should be included within a manifest.
+     */
+    isAddressable?: boolean;
+    /**
+     * Whether the component requires the parent to be present when deploying/retrieving
+     */
+    unaddressableWithoutParent?: boolean;
+
+    /**
+         * Whether or not components of the same type can be can be specified with the wildcard character, and by name in a manifest
+         *
+         ```
+         <members>*</members>
+         <members>Account</members>
+         <name>CustomObject</name>
+         ```
+         */
+    supportsWildcardAndName?: boolean;
+
+    /**
+     * Whether the component can be partially deleted, such as metadata types that are made up of multiple files.
+     *
+     * __Examples:__ `LightningComponentBundle`, `ExperienceBundle`, `StaticResource`, and `DigitalExperienceBundle`
+     */
+    supportsPartialDelete?: boolean;
+
+    /**
+     * Whenever this type is requested, return the aliasFor type instead
+     */
+    aliasFor?: string;
+    /**
+     * Type definitions for child types, if the type has any.
+     *
+     * __Examples:__ `CustomField` and `CompactLayout` on `CustomObject`
+     */
+    children?: {
+      types: TypeIndex;
+      suffixes: SuffixIndex;
+      directories?: DirectoryIndex;
+    };
+    /**
+     * Configuration for resolving and converting components of the type.
+     */
+    strategies?: {
+      adapter:
+        | "mixedContent"
+        | "matchingContentFile"
+        | "decomposed"
+        | "digitalExperience"
+        | "bundle"
+        | "default";
+      transformer?:
+        | "decomposed"
+        | "staticResource"
+        | "nonDecomposed"
```

Comment:
I think I'd like to NOT expose the stuff that CustomLabels was using.  I don't want more people creating dependencies on those behaviors since they've been so bad.